### PR TITLE
Test-Driven-Development

### PR DESCRIPTION
**Pull request by vkondrashkov**
### Note
Actually tests were set up before writing code, but haven't commited it

### Tests
<table style="width:100%">
  <tr>
    <th>Before</th>
    <th>After</th> 
  </tr>
  <tr>
    <td>before</td>
    <td>after</td> 
  </tr>
</table>

### CODE REVIEW

File: eStore/Core/Networking/eStore.swift [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+//  eStore
+//
+//  Created by Vladislav Kondrashkov on 11/17/19.
+//  Copyright © 2019 Vladislav Kondrashkov. All rights reserved.
+//
+
+import Moya
+
+enum eStore {
+    case getSmartphones
```
Comment (pkandrashkou) [inline eStore/Core/Networking/eStore.swift]: Are you Java man? get? LUL

File: eStore/Core/Utils/Result+Values.swift [hunk 2]
```diff
@@ -0,0 +1,25 @@
+//
+//  Result+Values.swift
+//  eStore
+//
+//  Created by Vladislav Kondrashkov on 11/17/19.
+//  Copyright © 2019 Vladislav Kondrashkov. All rights reserved.
+//
+
+import Foundation
```
Comment (pkandrashkou) [inline eStore/Core/Utils/Result+Values.swift]: Why import foundation here?

File: eStore/Core/Networking/ProductsRepository.swift [hunk 3]
```diff
@@ -0,0 +1,110 @@
+//
+//  ProductsRepository.swift
+//  eStore
+//
+//  Created by Vladislav Kondrashkov on 11/17/19.
+//  Copyright © 2019 Vladislav Kondrashkov. All rights reserved.
+//
+
+import Foundation
```
Comment (pkandrashkou) [inline eStore/Core/Networking/ProductsRepository.swift]: Hey Foundation, why are you here?

File: eStore/Core/Networking/ProductsRepository.swift [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+import Moya
+import ObjectMapper
+import Result
+
+enum ProductsRepositoryError: Error {
+    case badRequest
```
Comment (pkandrashkou) [inline eStore/Core/Networking/ProductsRepository.swift]: Why do you expose internals of repisoty implementation? Keep it simple - `invalidInput` or / and `failed`

File: eStore/Core/Networking/eStore.swift [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+            return makeStubData(name: "products-laptop-\(id)")
+        case .getTVs:
+            return makeStubData(name: "products-tvs")
+        case .getTV(let id):
+            return makeStubData(name: "products-tv-\(id)")
+        }
+    }
+
+    private func makeStubData(name: String) -> Data {
+        let stubUrl = Bundle.main.url(forResource: "\(name)-stub", withExtension: "json")
```
Comment (pkandrashkou) [inline eStore/Core/Networking/eStore.swift]: Check this out.
https://github.com/artsy/eidolon/blob/master/Kiosk/App/Networking/ArtsyAPI.swift#L418

stubbed jsons are part of test, not the app

File: eStore/Core/Networking/eStore.swift (hunk 1 above)
Comment (vkondrashkov) [reply eStore/Core/Networking/eStore.swift]: It’s made for differences between request, then I’ll make post and delete ones

Comment (pkandrashkou) [reply eStore/Core/Networking/eStore.swift]: This is not Swift style at all. Just write `.smartphones` when you wanna get smartphones. Wanna delete them? Say it explicitly. `.deleteSmartphone(id: String)`

File: eStore/Core/Networking/ProductsRepository.swift (hunk 3 above)
Comment (pkandrashkou) [reply eStore/Core/Networking/ProductsRepository.swift]: image

**Pull request merged**